samples/azure-function/README.md

```diff
@@ -0,0 +1,121 @@
+# Use Carbon Aware SDK with an Azure Function
+
+## Overview
+
+The sample included showcase the Azure Functions tooling for the Carbon Aware SDK [C# Class Library](../../docs/architecture/c-sharp-client-library.md). It includes an implementation for `GetAverageCarbonIntensity` and for `GetCurrentForecast`. `GetAverageCarbonIntensity` uses the `EmissionsHandler` to return the carbon intensity rate of a location for a specific timespan. `GetCurrentForecast` uses the `ForecastHandler` to yield the optimal time of a specified location and duration. The functions can run locally for debugging or be deployed to Azure.
+
+## Azure Function Dependency Injection
+
+The Carbon Aware SDK is included in the function .csproj file by [creating and adding the SDK as a package](../../docs/packaging.md#included-scripts).  The [Startup.cs](./Startup.cs) file uses dependency injection to access the handlers in the library. The following code initializes the C# Library:
+
+```C#
+ public override void Configure(IFunctionsHostBuilder builder)
+        {
+            var configuration = builder.GetContext().Configuration;
+            builder.Services
+            .AddEmissionsServices(configuration)
+            .AddForecastServices(configuration);
+        }
+```
+
+## Run Function Locally
+
+Both Azure Function apps can be run locally without needing an azure subscription. The process for running both is the same, as they use the same configuration, just call different paths within the SDK.
+
+### Prerequisites to Run
+
+[.NET Core SDK](https://dotnet.microsoft.com/download)
+
+[Azure Functions Core Tools](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local)
+
+### Start Function
+
+To run and debug locally, update the [appsettings.json](appsettings.json) file to include the desired [configuration](../../docs/configuration.md).
+
+In the app folder (`samples/azure-function`), run the command:  ```func start```
+
+After the function has compiled and is running, the URLs to the functions will be presented.  
+
+#### _Example call for Get Average Carbon Intensity function_
+
+The following example will retrieve the Average Carbon Intensity.  For this example, query parameters were used, but the values could also be sent in the body of the request.
+
+```bash
+curl --location --request GET 'http://localhost:7071/api/GetAverageCarbonIntensity?startDate=2022-03-01T15:30:00Z&endDate=2022-03-01T18:30:00Z&location=eastus'
+```
+
+#### _Example call for Get Current Forecast function_
+
+The following example will call the Current Forecast route.  If an error is returned, update the start and end dates.  The request can use either the request body or query parameters.
```

Comment:
```suggestion
The following example will call the Current Forecast route.  If an error is returned, update the start and end dates.  The request can use either the request body or query parameters.
```
Can you be more specific about updating the start and end dates? I assume it is due to the dates has to be in the future?

Reply:
Yeah, this is detailed in the readme docs but I'll make a note here for quick reference